src/components/plots/ContinuousEmbeddingPlot.jsx

```diff
@@ -89,7 +89,11 @@ const ContinuousEmbeddingPlot = (props) => {
       );
     }
 
-    if (!config || loading || embeddingLoading || !plotComponent) {
+    if (!config
+      || loading
+      || embeddingLoading
+      || !plotComponent
+      || Object.keys(plotSpec).length === 0) {
```

Comment:
why do we need this check: `Object.keys(plotSpec).length === 0`?

Reply:
This is the cause of the bug. An empty plot spec renders an empty plot, so this conditional makes sure that plot spec is loaded before rendering a Vega plot.